microsoft/testsuites/network/networksettings.py

```diff
@@ -563,75 +567,117 @@ def validate_device_statistics(self, environment: Environment, log: Logger) -> N
+                    ]
+
+                    per_rx_queue_packets = [
+                        v
+                        for (k, v) in device_stats.counters.items()
+                        if self._rx_queue_stats_regex.search(k)
+                    ]
+
+                if len(per_tx_queue_packets) and len(per_rx_queue_packets) < 4:
+                    raise SkippedException(
+                        "Skipping the test as the number of queues are less than 4."
+                    )
 
-                per_rx_queue_packets = [
-                    v
-                    for (k, v) in device_stats.counters.items()
-                    if self._vf_rx_stats_regex.search(k)
-                ]
-            else:
-                per_tx_queue_packets = [
-                    v
-                    for (k, v) in device_stats.counters.items()
-                    if self._tx_queue_stats_regex.search(k)
+                starving_tx_queues = [
+                    queue
+                    for queue, pkts in enumerate(per_tx_queue_packets)
+                    if pkts == 0
                 ]
 
-                per_rx_queue_packets = [
-                    v
-                    for (k, v) in device_stats.counters.items()
-                    if self._rx_queue_stats_regex.search(k)
+                starving_rx_queues = [
+                    queue
+                    for queue, pkts in enumerate(per_rx_queue_packets)
+                    if pkts == 0
                 ]
 
-            avg_tx_queue_packets = sum(per_tx_queue_packets) / len(per_tx_queue_packets)
-            per_tx_queue_pkt_percent: List[float] = []
-            for packet_count in per_tx_queue_packets:
-                per_tx_queue_pkt_percent.append(packet_count / avg_tx_queue_packets)
+                starved_queues_per_device[device_stats.interface] = (
+                    starving_tx_queues,
+                    starving_rx_queues,
+                )
 
-            assert_that(
-                (max(per_tx_queue_pkt_percent) - min(per_tx_queue_pkt_percent)),
-                f"{an_enabled}Statistics show traffic is not evenly distributed"
-                " among tx queues",
-            ).is_less_than_or_equal_to(0.5)
+            for k in starved_queues_per_device:
+                if k in prev_starved_queues_per_device:
+                    txq_list, rxq_list = starved_queues_per_device[k]
+                    prev_txq_list, prev_rxq_list = prev_starved_queues_per_device[k]
+                    final_starved_txq = [
+                        txq for txq in txq_list if txq in prev_txq_list
+                    ]
+                    final_starved_rxq = [
+                        rxq for rxq in rxq_list if rxq in prev_rxq_list
+                    ]
+
+            if (
+                not prev_starved_queues_per_device
+                and (not final_starved_txq)
+                and (not final_starved_rxq)
+            ):
+                raise PassedException(
```

Comment:
Doesn't raise PassException, if it's fully passed. The Pass exception is used for "partial" passed or pass with warning.

Reply:
hmm I have to make the test pass form here. How do I do it explicitly if not with passed exception?

Reply:
It can be return directly. But make sure the comments is clear enough. The test logic is not easy to understand, when it's returning as pass in the middle of code.